platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -760,7 +933,57 @@ namespace platoon_strategic_ihp
                 ROS_DEBUG_STREAM("Publishing request to the leader " << senderId << " with params " << request.strategy_params << " and plan id = " << request.header.plan_id);
                 potentialNewPlatoonId_front = newLeaderPlatoonId;
             }
-            
+
+            // UCLA: Condition passed the check for cut-in from front
+            else if (isVehicleNextLane(frontVehicleDtd, rearVehicleDtd, frontVehicleCtd))
+            {
+                /**
+                 * UCLA Implementation note:
+                 *  1. isVehicleNextLane --> determine if the platoon is next to host vheicle
+                 *  2. sender_id == requesting veh ID --> front joiner ID
+                 */
+                std::string newLeaderPlatoonId = pm_.currentPlatoonID; // set self's (front join veh) platoon ID as the new platoon ID
+                ROS_DEBUG_STREAM("Host vehicle platoon id = " << newLeaderPlatoonId << "; join the rear platoon as the new leader (frontal join).");
+                
+                // compose request
+                cav_msgs::MobilityRequest request;
+                request.header.plan_id = boost::uuids::to_string(boost::uuids::random_generator()());
+                request.header.recipient_id = senderId;
+                request.header.sender_id = config_.vehicleID;
+                request.header.timestamp = ros::Time::now().toNSec() / 1000000;
+                request.location = pose_to_ecef(pose_msg_);
+                // UCLA: Desired joining index for cut-in join, indicate the index of gap-leading vehicle. -1 indicate cut-in from front.
+                request.join_index = -1; // Note: not used by same-lane functions.
```

Comment:
The isVehicleNextLane() method that got us into this block returns true if host is anywhere beside the platoon. not necessarily near the front. So how can we now assume that this will be a front join?